localstack-core/localstack/services/cloudformation/engine/v2/change_set_model_preproc.py

```diff
@@ -304,6 +308,10 @@ def _resolve_reference(self, logical_id: str) -> PreprocEntityDelta:
         node_parameter = self._get_node_parameter_if_exists(parameter_name=logical_id)
         if isinstance(node_parameter, NodeParameter):
             parameter_delta = self.visit(node_parameter)
+            if isinstance(parameter_delta.before, ResolvedParameter):
```

Comment:
Why do we need to first wrap the relevant values in a ResolvedParameter class and only evaluate them here?